spec/models/address_spec.rb

```diff
@@ -0,0 +1,111 @@
+require 'rails_helper'
+
+RSpec.describe Address, type: :model do
+  it 'is valid with valid attributes' do
+    address = Address.new(
+      country: 'Brazil',
+      state: 'Rio de Janeiro',
+      city: 'Rio de Janeiro',
+      neighbourhood: 'Copacabana',
+      street: 'Barata Ribeiro',
+      number: 100
+    )
+    expect(address).to be_valid
+  end
+
+  it 'is not valid without a country' do
+    address = Address.new(
+      country: nil,
+      state: 'Rio de Janeiro',
+      city: 'Rio de Janeiro',
+      neighbourhood: 'Copacabana',
+      street: 'Barata Ribeiro',
+      number: 100
+    )
+    expect(address).to_not be_valid
+  end
+
+  it 'is not valid without a state' do
+    address = Address.new(
+      country: 'Brazil',
+      state: nil,
+      city: 'Rio de Janeiro',
+      neighbourhood: 'Copacabana',
+      street: 'Barata Ribeiro',
+      number: 100
+    )
+    expect(address).to_not be_valid
+  end
+
+  it 'is not valid without a city' do
+    address = Address.new(
+      country: 'Brazil',
+      state: 'Rio de Janeiro',
+      city: nil,
+      neighbourhood: 'Copacabana',
+      street: 'Barata Ribeiro',
+      number: 100
+    )
+    expect(address).to_not be_valid
+  end
+
+  it 'is not valid without a neighbourhood' do
+    address = Address.new(
+      country: 'Brazil',
+      state: 'Rio de Janeiro',
+      city: 'Rio de Janeiro',
+      neighbourhood: nil,
+      street: 'Barata Ribeiro',
+      number: 100
+    )
+    expect(address).to_not be_valid
+  end
+
+  it 'is not valid without a street' do
+    address = Address.new(
+      country: 'Brazil',
+      state: 'Rio de Janeiro',
+      city: 'Rio de Janeiro',
+      neighbourhood: 'Copacabana',
+      street: nil,
+      number: 100
+    )
+    expect(address).to_not be_valid
+  end
+
+  it 'is not valid without a number' do
+    address = Address.new(
+      country: 'Brazil',
+      state: 'Rio de Janeiro',
+      city: 'Rio de Janeiro',
+      neighbourhood: 'Copacabana',
+      street: 'Barata Ribeiro',
+      number: nil
+    )
+    expect(address).to_not be_valid
+  end
+
+  it 'is not valid if number is not an integer' do
+    address = Address.new(
+      country: 'Brazil',
+      state: 'Rio de Janeiro',
+      city: 'Rio de Janeiro',
+      neighbourhood: 'Copacabana',
+      street: 'Barata Ribeiro',
+      number: 'Powerful Place'
+    )
+    expect(address).to_not be_valid
+  end
+
+  it 'is not valid if number is negative' do
+    address = Address.new(
+      country: 'Brazil',
+      state: 'Rio de Janeiro',
+      city: 'Rio de Janeiro',
+      neighbourhood: 'Copacabana',
+      street: 'Barata Ribeiro',
+      number: -100
+    )
+    expect(address).to_not be_valid
+  end
```

Comment:
[OPTIONAL]

These tests are good but consider maybe you are not really testing if the models work as you expect. Whether it's valid or not valid does not actually say it works the way you expect when it is not valid. In this project it's fine, but for bigger projects getting more detailed may help catch bugs early. What do you expect to happen when it's not valid? What message do you expect to be displayed? You'd be surprised how often the message you expect to be returned is not what you get.

This is definitely OPTIONAL, since it may cause you to think differently about what you put in model, but it's just something to think about in general for tests.